### DISCUSSION TITLE
Hombrew install M1 packages for cross-compile.

### DISCUSSION THREAD

**Discussion by yikerman** [Homebrew/discussions#1964] [Everyday usage] [answered]
Is there a way to install M1 packages on Intel machines? I want this feature to build a C project into a universal executable.

What I expect is like

```bash
$ brew install $dep
$ brew install $dep using M1 bottles keg-only.
Build my project.
Cross-compile my project using the M1 formulas.
lipo the two executables.
```

Or is there a better way to handle this? Doing this is mainly because GitHub Action doesn’t have macOS runners with M1 chips.

**Comment by carlocab** [comment, marked as answer]
This isn't supported natively by Homebrew. You can potentially patch `brew` to pour M1 bottles on Intel, but you're on your own with doing that, I'm afraid.

**Reply by yikerman** [reply]
Can I manual download the bottles?

I’ve checked the GitHub Packages but why are the bottles containers?

**Reply by alebcay** [reply]
You can specify the desired architecture as a bottle tag when running `brew fetch`. For instance, `brew fetch --bottle-tag=arm64_big_sur libpng`.

**Reply by carlocab** [reply]
Note that these bottles will have the install names all wrong, so you'll need to also make liberal use of `install_name_tool` if you want to use them. ARM bottles will also have been built for a different prefix, so trying to install it outside of `/opt/homebrew` might make things a bit weird.
